server/src/main/java/io/crate/expression/scalar/regex/RegexpReplaceFunction.java

```diff
@@ -89,35 +94,32 @@ public Signature boundSignature() {
 
     @Override
     public Symbol normalizeSymbol(Function function, TransactionContext txnCtx) {
-        List<Symbol> arguments = function.arguments();
-        final int size = arguments.size();
+        List<Symbol> args = function.arguments();
+        final int size = args.size();
         assert size == 3 || size == 4 : "function's number of arguments must be 3 or 4";
-
-        if (anyNonLiterals(arguments)) {
+        if (anyNonLiterals(args)) {
             return function;
         }
-
-        final Input input = (Input) arguments.get(0);
-        final Input pattern = (Input) arguments.get(1);
-        final Input replacement = (Input) arguments.get(2);
-        final String inputValue = (String) input.value();
-        final String patternValue = (String) pattern.value();
-        final String replacementValue = (String) replacement.value();
-        if (inputValue == null || patternValue == null || replacementValue == null) {
+        final String value = ((Input<String>) args.get(0)).value();
+        final String pattern = ((Input<String>) args.get(1)).value();
+        final String replacement = ((Input<String>) args.get(2)).value();
```

Comment:
Are all arguments `String`? If so the generic definition could be changed, eliminating the casts here.